test/view/broadcast/broadcast_game_screen_test.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:chessground/chessground.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:http/testing.dart';
+import 'package:lichess_mobile/src/model/common/id.dart';
+import 'package:lichess_mobile/src/network/http.dart';
+import 'package:lichess_mobile/src/network/socket.dart';
+import 'package:lichess_mobile/src/view/broadcast/broadcast_game_screen.dart';
+import 'package:lichess_mobile/src/view/broadcast/broadcast_player_widget.dart';
+
+import '../../network/fake_websocket_channel.dart';
+import '../../test_helpers.dart';
+import '../../test_provider_scope.dart';
+
+final client = MockClient((request) {
+  if (request.url.path == '/api/broadcast/-/-/$_roundId') {
+    return mockResponse(
+      _roundResponse,
+      200,
+      headers: {'content-type': 'application/json; charset=utf-8'},
+    );
+  }
+  if (request.url.path == '/api/study/$_roundId/$_gameId.pgn') {
+    return mockResponse(_gameResponse, 200, headers: {'content-type': 'application/x-chess-pgn'});
+  }
+  return mockResponse('', 404);
+});
+
+void main() {
+  group('Broadcast game screen', () {
+    testWidgets('Displays broadcast game screen', variant: kPlatformVariant, (tester) async {
+      final fakeSocket = FakeWebSocketChannel();
+      final app = await makeTestProviderScopeApp(
+        tester,
+        home: const BroadcastGameScreen(
+          tournamentId: _tournamentId,
+          roundId: _roundId,
+          gameId: _gameId,
+        ),
+        overrides: [
+          lichessClientProvider.overrideWith((ref) => LichessClient(client, ref)),
+          webSocketChannelFactoryProvider.overrideWith((ref) {
+            return FakeWebSocketChannelFactory((_) => fakeSocket);
+          }),
+        ],
+      );
+
+      await tester.pumpWidget(app);
+
+      expect(find.byType(CircularProgressIndicator), findsOneWidget);
+
+      await tester.pump();
+
+      expect(find.byType(Chessboard), findsOneWidget);
+      expect(find.byType(BroadcastPlayerWidget), findsNWidgets(2));
+      expect(find.text('Dastan, Muhammed Batuhan'), findsOne);
+      expect(find.text('Gokerkan, Cem Kaan'), findsOne);
+    });
+
+    testWidgets('Receives a new move of the game', variant: kPlatformVariant, (tester) async {
+      final fakeSocket = FakeWebSocketChannel();
+      final app = await makeTestProviderScopeApp(
+        tester,
+        home: const BroadcastGameScreen(
+          tournamentId: _tournamentId,
+          roundId: _roundId,
+          gameId: _gameId,
+        ),
+        overrides: [
+          lichessClientProvider.overrideWith((ref) => LichessClient(client, ref)),
+          webSocketChannelFactoryProvider.overrideWith((ref) {
+            return FakeWebSocketChannelFactory((_) => fakeSocket);
+          }),
+        ],
+      );
+
+      await tester.pumpWidget(app);
+
+      await tester.pump();
+      // await tester.pump(); It might be interesting to know why it breaks the test
```

Comment:
remove the comment, if you have questions or things you don't understand it is better to leave the comment directly in GitHub PR page as long as the review is not done.

Reply:
I am trying to find the cause of the issue. The reason it breaks the test is that after receiving a move the screen is in loading state again.

I also noticed yesterday watching the world championship that the screen blinks each time a move is received.

Reply:
This seems to be a consequence of using `selectAsync` to get the `broadcastRoundController` state